### PR TITLE
Upgrade to latest packages incl Webpack 2.2

### PR DESCRIPTION
**Pull request by sickboy**
- Update to latest Webpack, latest tslint, switch to awesome-typescript-loader, ExtractTextPlugin
- Add vscode config

### CODE REVIEW

File: config/webpack/dev.js [hunk 1]
```diff
@@ -1,20 +1,22 @@
+var fs = require('fs');
 var path = require('path');
 var webpack = require('webpack');
 var postcssAssets = require('postcss-assets');
 var postcssNext = require('postcss-cssnext');
 var stylelint = require('stylelint');
 var ManifestPlugin = require('webpack-manifest-plugin');
 
+const {
```
Comment (altayaydemir) [inline config/webpack/dev.js]: Seems like it breaks the build https://travis-ci.org/barbar/vortigern/builds/193975281

Comment (altayaydemir) [inline config/webpack/dev.js]: I guess its because we use node 5 in Travis config.

Comment (sickboy) [inline config/webpack/dev.js]: Yea, I fixed it now with latest commit, incl more test fixes.

Comment (altayaydemir) [inline config/webpack/dev.js]: can you open another PR?

Comment (sickboy) [inline config/webpack/dev.js]: I cherry picked it and merged it now on develop https://github.com/barbar/vortigern/commit/7961565a93571aea94517d9804e49a489d0a4a8b

Comment (altayaydemir) [inline config/webpack/dev.js]: hah, I've just seen your commit. its ok. thanks 👍

Comment (sickboy) [inline config/webpack/dev.js]: No worries. I should've ran the tests before.

I really dislike the need to manage multiple copies/versions of the webpack configs, 
it's cumbersome and error prone. These guys did quite an amazing job, it's used in Aurelia: https://github.com/easy-webpack
Perhaps it's interesting to look into.

**Pull request merged**